polar-core/src/polar.rs

```diff
@@ -1,17 +1,15 @@
 use super::error::PolarResult;
-use super::formatting::source_lines;
 use super::parser;
 use super::rewrites::*;
 use super::types::*;
 use super::vm::*;
+use super::warnings::check_singletons;
 
-use std::collections::{hash_map::Entry, HashMap};
-use std::io::{stderr, Write};
 use std::sync::{Arc, RwLock};
 
 pub struct Query {
-    vm: PolarVirtualMachine,
-    done: bool,
+    pub vm: PolarVirtualMachine,
```

Comment:
what do you think about exposing a pub method on `Query` like `get_next_message()` instead of making the `vm` property pub?